compiler/src/dotty/tools/dotc/core/Symbols.scala

```diff
@@ -540,6 +540,11 @@ object Symbols {
       d != null && d.flagsUNSAFE.is(Private)
     }
 
+    /** Is the symbol a pattern bound symbol?
+     */
+    final def isPatternBound(implicit ctx: Context): Boolean =
+      !isClass && this.is(Case, butNot = Enum | Module)
```

Comment:
```suggestion
      !isClass && is(Case, butNot = Enum | Module)
```

Reply:
Initially, I tried `is(...)`, but it does not check --- it makes sense, as adaptation only happens with a select.

Reply:
Ok